tapo4el/aws-backoffice
# Task 9

3 - A working and correct Express application should be in the bff-service folder. Reviewer can start this application locally with any valid configuration in the .env file and this application should works as described in the Task 9.1 - **DONE**
5 - The BFF Service should be deployed with Elastic Beanstalk. The BFF Service call should be redirected to the appropriate service : Product Service or Cart Service. The response from the BFF Service should be the same as if Product Service or Cart Service services were called directly. - **DONE**

+1 - Add a cache at the BFF Service level for a request to the getProductsList lambda function of the Product Service. The cache should expire in 2 minutes. - **DONE**
+1 - Use NestJS to create BFF Service instead of Express - **DONE**

UI changes: https://github.com/tapo4el/shop-react-redux-cloudfront/pull/5

direct Product API: https://wyofw9a4rb.execute-api.us-east-1.amazonaws.com/products
bff Product API: http://tapo4el-bff-api-dev.us-east-1.elasticbeanstalk.com/products

direct Cart API: http://tapo4el-cart-api-dev.us-east-1.elasticbeanstalk.com/api/profile/cart
bff Cart API: http://tapo4el-bff-api-dev.us-east-1.elasticbeanstalk.com/profile/cart

How to test:
1) Open http://d3daouilse61ba.cloudfront.net/ (please open in INCOGNITO mode to skip the cached version in your browser)
2) Go to "Manage products" section in the drop-down menu in the site header
3) Push the button "Create Product" and fill in the fields with necessary data
4) Go back to the main page and after the cache will expire your product will appear.

PS: Feel free to reach me in discord or Teams if you will have any questions.
